source/opt/inst_bindless_check_pass.h

```diff
@@ -28,18 +28,19 @@ namespace opt {
 // external design may change as the layer evolves.
 class InstBindlessCheckPass : public InstrumentPass {
  public:
-  // For test harness only
-  InstBindlessCheckPass()
-      : InstrumentPass(7, 23, kInstValidationIdBindless, 1),
-        input_length_enabled_(true),
-        input_init_enabled_(true) {}
-  // For all other interfaces
+  // Deprecated interface
   InstBindlessCheckPass(uint32_t desc_set, uint32_t shader_id,
                         bool input_length_enable, bool input_init_enable,
                         uint32_t version)
       : InstrumentPass(desc_set, shader_id, kInstValidationIdBindless, version),
         input_length_enabled_(input_length_enable),
         input_init_enabled_(input_init_enable) {}
```

Comment:
Should we issue an error if the version is not 2?  If somebody is still trying to get version 1, they should be told.

Reply:
I did add an assert in instrument_pass.cpp at the top of InstrumentInitialize() which should be sufficient to catch any uses of version 1 when they test the layers. Again, once the layers transition away from specifying the version, the "version" interfaces will go away as well as version_ and the assert. So I think we are covered. Let me know if you still see a hole.

Reply:
Okay, now I see the assert.  The assert will not trigger in a release build.  Could we log an error message, which will be seen with release builds?

Reply:
OK. Will do.